### PR TITLE
[.NET] Add support for parsing schema 1.4 features

### LINKED ISSUE #5503: [.NET] Support parsing of Action.Execute

### LINKED ISSUE #5508: [WPF] Render Action.Execute

### PR DESCRIPTION
**Pull request by RebeccaAnne**
## Related Issue
Fixes #5503
Fixes #5508 

## Description
Add Parsing for schema 1.4 features, including basic handling of Action.Execute:
- Added Refresh property to AdaptiveCard
- Added Authentication property to AdaptiveCard
- Added Action.Execute action type. Updated behavior to behave identically to Action.Submit.
- Updated .NET visualizer to handle Action.Execute by showing the data/verb in a dialog
- Tests cases for parsing of Refresh/Authenticate

## Sample Card
Added new sample at [Action.Execute.json](https://github.com/microsoft/AdaptiveCards/blob/main/samples/v1.4/Elements/Action.Execute.json)

## How Verified
- Ran existing .NET tests
- Added new tests for parsing of Refresh and Authentication
- Confirmed Action.Execute sample behaved as expected in visualizer

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/5531)

### CODE REVIEW

File: source/dotnet/Library/AdaptiveCards/AdaptiveAuthentication.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        [XmlElement(typeof(AdaptiveTokenExchangeResource))]
+#endif
+        public AdaptiveTokenExchangeResource TokenExchangeResource { get; set; }
+
+        /// <summary>
+        ///     Buttons that should be displayed to the user when prompting for authentication. The array MUST contain one button of type \"signin\". Other button types are not currently supported.
+        /// </summary>
+        /// <summary>
+        /// Collection of images to display.
+        /// </summary>
```
Comment (paulcam206) [inline source/dotnet/Library/AdaptiveCards/AdaptiveAuthentication.cs:43]: looks like this snuck in :)

**Review by paulcam206 — APPROVED** [review]
only one minor doc issue

:shipit:

**Review by jwoo-msft — APPROVED** [review]

File: source/dotnet/Library/AdaptiveCards/Rendering/AdaptiveVisitor.cs [hunk 2]
```diff
@@ -164,6 +167,9 @@ public virtual void Visit(AdaptiveShowCardAction action)
         public virtual void Visit(AdaptiveToggleVisibilityAction action)
         {
         }
+        public virtual void Visit(AdaptiveExecuteAction media)
```
Comment (jwoo-msft) [inline source/dotnet/Library/AdaptiveCards/Rendering/AdaptiveVisitor.cs:170]: nit: media to action or executeAction?

**Review by jwoo-msft — COMMENTED** [review]
I don't see AdaptiveAction changes.

**Review by paulcam206 — APPROVED** [review]

**Review by golddove — APPROVED** [review]

**Review by jwoo-msft — APPROVED** [review]

**Pull request merged**